components/profile_popover/profile_localtime.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {DateTime, Duration} from 'luxon';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserTimezone} from '@mattermost/types/users';
+import Timestamp from 'components/timestamp';
+
+type ProfileTimezoneProps = {
+    profileUserTimezone?: UserTimezone;
+    currentUserTimezone: string | undefined;
+}
+
+const returnTimeDiff = (
+    currentUserTimezone: string | undefined,
+    profileUserTimezone: UserTimezone | undefined,
+) => {
+    if (!currentUserTimezone || !profileUserTimezone) {
+        return undefined;
+    }
+    const currentUserDate = DateTime.local().setZone(currentUserTimezone);
+    const profileUserDate = DateTime.local().setZone(profileUserTimezone?.manualTimezone || profileUserTimezone?.automaticTimezone);
+
+    const offset = Duration.fromObject({
+        hours: (profileUserDate.offset - currentUserDate.offset) / 60,
+    });
+
+    if (!offset.valueOf()) {
+        return undefined;
+    }
+
+    const timeOffset = offset.toHuman({unitDisplay: 'short', signDisplay: 'never'});
+
+    return offset.valueOf() > 0 ? (
+        <FormattedMessage
+            id='user_profile.account.hoursAhead'
+            defaultMessage='({timeOffset} ahead)'
+            values={{timeOffset}}
+        />
+    ) : (
+        <FormattedMessage
+            id='user_profile.account.hoursBehind'
+            defaultMessage='({timeOffset} behind)'
+            values={{timeOffset}}
+        />
+    );
+};
+
+const ProfileTimezone = (
+    {
+        currentUserTimezone,
+        profileUserTimezone,
+    }: ProfileTimezoneProps,
+) => {
+    const profileTimezone = profileUserTimezone?.manualTimezone || profileUserTimezone?.automaticTimezone;
```

Comment:
Would you mind using `getUserCurrentTimezone` instead? `manualTimezone` can have a value even when `UserTimezone['useAutomaticTimezone']` is set to true.

Rough adaptation from an existing example:
```suggestion
    const profileTimezone = getUserCurrentTimezone(profileUserTimezone) ?? undefined
```

Reply:
So, `profileUsertimeZone` is the timezone of the user you are viewing in the UI, not the user who is doing the viewing. Ex: `profileUserTimezone` is you, and `currentUserTimezone` is me if i'm the one looking at you.

With that said, wouldn't the `getUserCurrentTimezone` pull me instead of you?

Reply:
`getUserCurrentTimezone` is named a bit odd, but no—it will pull the proper timezone for the `UserTimezone` config you pass it. All it does is determine if the auto or manual timezone should be used according to `automaticTimezone === 'true'`—using it just keeps the conditional logic standardized.

Reply:
Oh cool, done!

Reply:
`getUserCurrentTimezone` is the one that processes a `UserTimezone`, and `getCurrentUserTimezone` is the one that will select the `currentUser's` timezone from redux

Reply:
Resolved by fetching the profile user with `getUserCurrentTimezone`